src/auth/aws.js

```diff
@@ -39,12 +40,112 @@ export function createCanonicalRequest(request) {
         CanonicalQueryString: _.encodeQueryStringParameters(url.searchParams),
         CanonicalHeaders: _.formatCanonicalHeaders(headers),
         SignedHeaders: _.formatSignedHeaders(headers),
-        HashedPayload: hex(sha256(body)),
+        HashedPayload: hex(_.toSHA256(body)),
     };
 
     return `${CanonicalRequestObj.HTTPMethod}\n${CanonicalRequestObj.CanonicalURI}\n${CanonicalRequestObj.CanonicalQueryString}\n${CanonicalRequestObj.CanonicalHeaders}\n${CanonicalRequestObj.SignedHeaders}\n${CanonicalRequestObj.HashedPayload}\n`;
 }
 
-export function createStringToSign() {
+/**
+ * Creates Scope object for signed string i.e. "20130606/us-east-1/s3/aws4_request"
+ * 
+ * @export
+ * @returns {String}
+ */
+export function createScope(awsRegion) {
+    return `${_.getShortDate()}/${awsRegion}/s3/aws4_request`;
+}
+
+
+/**
+ * Returns string to sign
+ * 
+ * @export
+ * @param {any} canonicalRequest 
+ * @returns 
+ */
+export function createStringToSign(canonicalRequest, scope) {
+    if (!canonicalRequest) throw Error(ERROR_PARAM_REQUIRED);
+    if (typeof canonicalRequest !== 'string') throw Error(ERROR_PARAM_TYPE_IS_NOT_STRING);
+
+    const timeStampISO8601Format = _.getAWSTimestamp();
+    const request = hex(_.toSHA256(canonicalRequest).toString());
+
+    return `${AUTH_IDENTIFIER_HEADER}\n${timeStampISO8601Format}\n${scope}\n${request}`;
+}
+
+/**
+ * Returns Signing key
+ * 
+ * @export
+ * @param {any} awsSecretKey 
+ * @param {any} awsRegion 
+ * @returns {String}
+ */
+export function createSigningKey(awsSecretKey, awsRegion) {
+    if (!awsSecretKey) throw Error(ERROR_PARAM_REQUIRED);
+    if (typeof awsSecretKey !== 'string') throw Error(ERROR_PARAM_TYPE_IS_NOT_STRING);
+    if (!awsRegion) throw Error(ERROR_PARAM_REQUIRED);
+    if (typeof awsRegion !== 'string') throw Error(ERROR_PARAM_TYPE_IS_NOT_STRING);
+
+    return _.toHmacSHA256(_.toHmacSHA256(_.toHmacSHA256(_.toHmacSHA256(`AWS4${awsSecretKey}`, _.getShortDate()), awsRegion), 's3'), 'aws4_request');
```

Comment:
you didn't update all these references. I was expecting you to update the order of the params in the `toHmacSHA256` method rather than in the one call, as (I _think_) you have used the same param order as in the AWS docs throughout. Now I think it's inconsistent...